File: bannerlayout/src/main/java/com/wkz/bannerlayout/annotation/ProgressShapeMode.java
package com.wkz.bannerlayout.annotation;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({
        ProgressShapeMode.RECTANGLE,
        ProgressShapeMode.RING
})
@Retention(RetentionPolicy.CLASS)
public @interface ProgressShapeMode {
    int RECTANGLE = 0;
    int RING = 1;
}

File: bannerlayout/src/main/java/com/fphoenixcorneae/bannerlayout/annotation/ProgressShapeMode.kt
package com.fphoenixcorneae.bannerlayout.annotation

import androidx.annotation.IntDef

@IntDef(
        ProgressShapeMode.RECTANGLE,
        ProgressShapeMode.RING
)
@kotlin.annotation.Retention(AnnotationRetention.BINARY)
annotation class ProgressShapeMode {
    companion object {
        const val RECTANGLE = 0
        const val RING = 1
    }
}